Oracle.md

```diff
@@ -0,0 +1,190 @@
+They use `R0` and `R1` to cover the cases where BTC/USD >= $20000 and < $99999, using `R0=0` and `R1 in [2,9]`.
+Finally, they use only `R0` to cover the cases where BTC/USD >= $100000 and <= $999999, using `R0 in [1-9]`.
+
+At maturity time, if the BTC/USD rate is less than $999999, the oracle signs each of the digits representing the outcome value using the digits' corresponding R value.
+If the rate is greater than $999999, the oracle signs the value $999999.
+
+### Serialization and signing of outcome values
+
+Every outcome value should be encoded as [a UTF-8 string with NFC normalization](./Messaging.md#Fundamental-types), hashed using sha256 before finally being signed by an oracle.
+UTF-8 is chosen as being a widely supported and easy to implement encoding format.
+
+For numerical outcomes represented in bases greater than 10, each digit should be converted to base 10 before being encoded (note that base 10 numbers in UTF-8 take values in the 0x0030-0x0039 range).
+This helps preventing any confusion about the capitalization of letters or the introduction of non-standard characters.
+
+### Serialization of event descriptors
+
+Event descriptors should be serialized using [TLV format](https://github.com/lightningnetwork/lightning-rfc/blob/master/01-messaging.md#type-length-value-format) as described bellow.
+
+#### Version 0 `enum_event_descriptor`
+
+1. type: 55302 (`enum_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`u16`:`num_outcomes`]
+   * [`u16`:`outcome_1_len`]
+   * [`string`:`outcome_1`]
+   * ...
+   * [`u16`:`outcome_n_len`]
+   * [`string`:`outcome_n`]
+
+This type of event descriptor is a simple enumeration where the value `n` is the number of outcomes in the event.
+
+Note that `outcome_i` is the outcome value itself and not its hash that will be signed by the oracle.
+
+#### Version 0 `range_event_descriptor`
+
+1. type: 55304 (`range_event_descriptor_v0`)
+2. data:
+   * [`x_point`:`oracle_nonce`]
+   * [`int32`:`start`]
+   * [`u32`:`count`]
+   * [`u16`:`step`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+
+#### Version 0 `large_range_event_descriptor`
+
+1. type: 55306 (`large_range_event_descriptor_v0`)
+2. data:
+   * [`bigsize`:`base`]
+   * [`bool`:`is_signed`]
+   * [`u16`:`nb_nonces`]
+   * [`nb_nonces*x_point`:`oracle_nonces`]
+   * [`string`:`unit`]
+   * [`int32`:`precision`]
+
+### Oracle events
+
+For users to be able to create DLCs based on a given event, they also need to obtain information about the oracle and the time at which it plans on releasing a signature over the event outcome.
+Oracle events contain such information, which includes:
+* the oracle public key,
+* the earliest time (UTC) at which it plans on releasing a signature over the event outcome, in epoch seconds,
+* the event descriptor,
+* the event URI which can be a name or categorization associated with the event by the oracle.
```

Comment:
I wonder if we should dissociate a oracle data transport + event identifier (e.g a HTTPS connection to a well-known server accessing an URL) from the concept of an oracle publication space. The former being an instance of the later class.

In the future we may have other oracle data transport, like federation of public boards or LN gossips infrastructure, thus increasing privacy and redundancy of communication between oracle and client, which would alleviate the concern raised at last meeting of an oracle "discovering" its signature consumers due to communication interactivity.

Reply:
This spec does not refer to communication layer at all, but the notion of URI can still be convenient to identify events I think. I had removed the part about URI specifications and HTTP examples because of what you mention. Maybe we should still have some specified ID scheme though, not sure.

Reply:
SGTM for now